polynote-kernel/src/main/scala/polynote/messages/Messages.scala

```diff
@@ -106,14 +129,82 @@ object CellResult extends MessageCompanion[CellResult](4)
 
 final case class ContentEdit(pos: Int, length: Int, content: String) {
   def applyTo(rope: Rope): Rope = if (length > 0) rope.delete(pos, length).insertAt(pos, Rope(content)) else rope.insertAt(pos, Rope(content))
+
+  // Given another edit which occured after this edit, create an edit that will be equivalent to applying this edit
+  // when applied after the given edit instead.
+  def rebase(other: ContentEdit): ContentEdit = other match {
+    // if the other edit is entirely before this edit, just shift this edit by the length delta
+    case ContentEdit(otherPos, otherLength, otherContent) if otherPos < pos && (otherPos + otherLength < pos) =>
```

Comment:
Could it be simplified to just `if otherPos + otherLength < pos =>`?

Reply:
Yep! Fixed